### PR TITLE
Underlining feature

### PR DESCRIPTION
**Pull request by lorenzopappone**
Fixed bug concerning the continuous underlining. When a misspelled word is manually edited by the user (using backspace), all the underlines are erased from the text. A new space character will trigger the underlines again.

### CODE REVIEW

File: flaskSpellChecker/static/underlineWords.js [hunk 1]
```diff
@@ -6,106 +6,171 @@ To include in your html code:
 
 */
 
+// class for text management
+class textClass {
```
Comment (mfdunlap) [inline flaskSpellChecker/static/underlineWords.js:10]: This looks really good! Really smart idea to make this a class.

File: flaskSpellChecker/static/underlineWords.js [hunk 2]
```diff
@@ -6,106 +6,171 @@ To include in your html code:
...
+  #htmlText;
+  #misspelledWords;
+
+  constructor(text = '', htmlText = '', misspelledWords = []) {
+    this.#text = text
+    this.#htmlText = htmlText
+    this.#misspelledWords = misspelledWords
+  }
+
+  getPlainText() {
```
Comment (mfdunlap) [inline flaskSpellChecker/static/underlineWords.js:22]: This has good function naming.

**Review by mfdunlap — APPROVED** [review]
Looks good!

File: flaskSpellChecker/static/underlineWords.js [hunk 3]
```diff
@@ -6,106 +6,171 @@ To include in your html code:
...
+          //console.log(text_value)
+          //console.log(data)
+          tmp = text_value
+
+          // check if misspelled word has been manually corrected using backspace
+          if (checkBackspace(e.key, data, tmp))
+            return
+
+          // update plain text (no html)
+          textSingleton.updateText(tmp)
```
Comment (manojv17) [inline flaskSpellChecker/static/underlineWords.js:82]: that's good idea of using singleton

File: flaskSpellChecker/static/underlineWords.js [hunk 4]
```diff
@@ -6,106 +6,171 @@ To include in your html code:
...
-        //alert(sel);
-        return sel.toString()
-    } 
-    else {
-        if (document.selection) {   // Internet Explorer before version 9
-            var textRange = document.selection.createRange();
-            return textRange.text
-        }
+          // update html text
+          textSingleton.updateHtmlText(document.getElementById("textArea").innerHTML)
```
Comment (manojv17) [inline flaskSpellChecker/static/underlineWords.js:98]: Good one, Singleton allows only single occurrence of itself to be created and provide access to that element itself.

**Pull request merged**